core/webgl-compatibility/src/Capabilities.ts

```diff
@@ -73,6 +73,24 @@ const buggyIntelMatchers = [
   /ANGLE \(Intel, Intel\(R\) (U)?HD Graphics 6(2|3)0 Direct3D11/,
 ];
 
+// Regexes to match as many Intel integrated GPUs as possible.
+// https://en.wikipedia.org/wiki/List_of_Intel_graphics_processing_units
+const integratedIntelGpuMatchers = [
+  /(U)?HD Graphics/,
+  /Iris (Pro|Plus|Xe)/,
+];
+
+function isIntegratedGraphics(args: {unmaskedVendor?: string, unmaskedRenderer?: string}): boolean {
+  if (args.unmaskedVendor && args.unmaskedVendor.includes("Intel")) {
+    if (args.unmaskedRenderer && integratedIntelGpuMatchers.some((x) => x.test(args.unmaskedRenderer!)))
+      return true;
+  }
+
+  // ###TODO: Detect AMD integrated graphics
+
```

Comment:
How should we categorize Apple M1 chipsets and mobile chipsets?  Technically these are integrated as well.

Reply:
See Paul's comment about the motivation being for devices possessing both discrete and integrated - so we should not count Apple M1 and mobile chipsets.